### ISSUE TITLE
Hebrew support

### ISSUE THREAD

**Issue by weitzhandler** [JetBrains/JetBrainsMono#119]
Hi,

I installed the font and like it a lot.
I live in Israel and a lot of my code, especially UIs and resources, contains Hebrew characters.
Please add support for Hebrew in this beautiful font.

Thanks

**Comment by DrorDvash** [comment]
I agree, Hebrew is required and will be great to have with all other languages.
thanks!

**Comment by Tomer51m** [comment]
Great font! but Hebrew support is essential.

**Comment by philippnurullin** [comment]
@weitzhandler Hi, we don't have expertise to do it. If you can suggest somebody who can help, it will be nice.

**Comment by mikeen** [comment]
The font is great, and Hebrew is in a desperate need for a good programming font.
[Alef](http://alef.hagilda.com/) is a great Hebrew open source font, one of it's creators @mushon might have the ability to assist.

The other programming font with Hebrew support is [PragmataPro](https://www.fsd.it/shop/fonts/pragmatapro/) by @fabrizioschiavi - he might be a "competitor" but maybe it can also be a lead.

Hope this gives some leads.

**Comment by ezity** [comment]
> If you can suggest somebody who can help, it will be nice.

@philippnurullin Sorry if this is a stupid question, but are you looking for someone to do this in a paid capacity?

The obvious parallel to JetBrains Mono would be [IBM Plex](https://github.com/IBM/plex), where they hired local type designers in order to expand the language support.

**Comment by philippnurullin** [comment]
@mikeen Thank you! This is really helpful.

@ezity I believe we don't need a outsource type designer for making the alphabet, but more of a consultant to look at result of our work & make corrections as a native speaker. But it's a subject to discuss.
The paid aspect can be discussed too, but i'm not the person to make that decision.

**Comment by ezity** [comment]
@philippnurullin As a Hebrew speaker (not native, but fluent) I am happy to give input. However, I am not a type designer so my comments may be of limited use.

I have the time available now to compile a list of Hebrew type designers, but I don't know how many of them would be willing to consult for free - which is why I ask about payment. If you think such a list would be helpful (of general Hebrew type designers - not specifically ones from open source projects) let me know and I'll do the research necessary.

Again, it may be a good idea to discuss with someone from the IBM Plex team as it is a similar project and they have made very good progress with non-latin language support.

**Comment by philippnurullin** [comment]
@ezity This will be really great! 
On the IBM Plex team, i don't see why they will help, but we will think of of as an option.

**Comment by ezity** [comment]
@philippnurullin 

- AlefAlefAlef designed [many fonts for brands](https://alefalefalef.co.il/en/resources/פונטים-לתאגידים/). Contact at fonts@alefalefalef.co.il.
- Yanek Iontef from Fontef collaborated on Hebrew design for [IBM Plex](https://www.ibm.com/plex/). Contact at info@fontef.com.
- Ben Nathan from Hafontia designed Hebrew for [Assistant](https://fonts.google.com/specimen/Assistant). Contact at hafontia@gmail.com.
- Fontbit - a studio of designers. Don't know about collaborations with brands but [this is their catalog](http://fontbit.co.il/2020/2020.pdf). Contact at info@fontbit.co.il.

I think the above would be a good place to start. If you need anything else let me know as I'd really like to see this font get Hebrew support.

**Comment by ezity** [comment]
I'd like to add one more:
- Hagilda cooperative collaborated on the design for the [open source font Alef](http://alef.hagilda.com). Contact at michal@michalsahar.co.il or danny@hatayas.co.il.

**Comment by weitzhandler** [comment]
@ezity would be wise for you to make sure they are happy of you posting their emails here in public.

**Comment by arielagor** [comment]
Hi — picking this thread back up five years in. I've built a **bridge font** that adds Hebrew coverage to JetBrains Mono today, and I'd like to offer it as a reference implementation + concrete starting point for an official designer-led effort.

### Repository

**https://github.com/arielagor/jetbrains-mono-hebrew** — JetBrains Mono Hebrew Bridge, SIL OFL 1.1.

### What shipped (v0.1)

- A Python merge pipeline that takes JetBrains Mono TTFs + Alef (Hagilda, OFL) and produces width-normalized Hebrew glyphs in the 600-unit mono cell.
- **49 Hebrew-adjacent glyphs** per font: 22 base letters, 5 final forms (ך ם ן ף ץ), 16 niqqud, 5 Hebrew punctuation marks, shekel (₪) — the last was previously missing from JetBrains Mono entirely.
- **4 weights** × TTF + WOFF2 (Regular, Bold, Italic, Bold-Italic).
- **GPOS \`mark\` feature** scoped to \`hebr\` script — niqqud attach to consonants via top/bottom/center anchors plus dedicated shin-dot/sin-dot positions.
- **GDEF** correctly classes niqqud as marks.
- Pixel-verified monospace alignment: 16 Latin chars and 16 Hebrew chars render to exactly the same width.
- 44-case pytest suite + GitHub Actions CI.
- Renamed "JetBrains Mono Hebrew Bridge" deliberately so users are clear this is community work, not official.

Specimen screenshots + full docs in the README.

### What it isn't

- Not a professional Hebrew design. It's Alef's Hebrew retrofitted into JB Mono's mono cell with a non-uniform scale (X=0.45, Y=0.65 — about 1.44x aspect distortion) to fit the widest glyph (shin) while keeping letter heights near Latin cap-height.
- No true italic Hebrew (Hebrew traditionally doesn't italicize — the Italic / Bold-Italic weights reuse Alef Regular / Bold on the Hebrew side).
- Anchors are shared per position class rather than per-letter-tuned.

### Why post it here

1. To give Hebrew-speaking devs something usable **today** while the proper design work happens.
2. To turn the official-design question into a concrete delta — "what would good look like?" is easier to answer against a shipping artifact than against a blank page, and the cost of a commission becomes "replace this bridge with designer output" rather than zero-to-one.

### Concrete offer

I'd be willing to co-fund a designer commission if JetBrains or the community wants to get to v1.0. The 2020 thread above surfaced four Hebrew foundries — [Fontef](http://fontef.com) (Yanek Iontef, did IBM Plex Sans Hebrew), [Hafontia](mailto:hafontia@gmail.com) (Ben Nathan, did Google's Assistant), [AlefAlefAlef](https://alefalefalef.co.il/en/), and [Hagilda](http://alef.hagilda.com) itself — any of whom would be natural choices. Happy to kick off outreach if there's interest from the maintainers in accepting the resulting PR upstream.

If Hebrew support upstream isn't on the roadmap but the bridge is useful, I'll keep maintaining it as a third-party distribution and point users here.

Either way — feedback on the bridge (especially from Hebrew-native devs) is welcome in the bridge repo's issue tracker.
